src/styles/calendar.scss

```
@@ -0,0 +1,60 @@
+$size: 100px;
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: 7 * $size + 7px;
+  margin: 50px auto;
+
+  div:nth-child(n) {
+    margin: 0 1px 1px 0;
+    background-color: #eee;
+    transition: background-color 0.5s, transform 0.5s;
+
+    &:hover {
+      background-color: #ffbfcb;
+      cursor: pointer;
+      transform: translate(0, -20px);
+    }
+  }
+
+  @for $i from 1 through 31 {
+    &-31 div:nth-child(#{$i})::before {
+      content: "#{$i}";
+      @include flex(flex,center,center);
+      width: $size;
+      height: $size;
+      border: 1px solid black;
+    }
+  }
+
+  @for $i from 1 through 30 {
+    &-30 div:nth-child(#{$i})::before {
+      content: "#{$i}";
+      @include flex(flex,center,center);
+      width: $size;
+      height: $size;
+      border: 1px solid black;
+    }
+  }
+
+  @for $i from 1 through 29 {
+    &-29 div:nth-child(#{$i})::before {
+      content: "#{$i}";
+      @include flex(flex,center,center);
+      width: $size;
+      height: $size;
+      border: 1px solid black;
+    }
+  }
+
+  @for $i from 1 through 28 {
+    &-28 div:nth-child(#{$i})::before {
+      content: "#{$i}";
+      @include flex(flex,center,center);
+      width: $size;
+      height: $size;
+      border: 1px solid black;
+    }
+  }
+}
```

Мені вже також писали чому щось там повторюється 4 рази в згенерованому файлі! Тому і тобі скажу що кожен твій цикл згенерує майже по 30 повторень кожного рядка що в ньому знаходиться!!!
Тому краще задай і розміри і рамки усім дівам одразу, там де ти фоновий колір та тривалість анімації прописав, а цим циклом тільки приховуй зайві клітинки через none.. Тоді тобі той міксин з однобуквенними змінними не знадобиться.)

Хоча тобі тоді доведеться зробити окремий цикл для контенту/цифр якщо тут зменшити ітерацію до 4-х разів. Але так напевне дубе краще, тимпаче можна count()/counter-inkrement використати..
```suggestion
  @for $i from 28 through 31 {
    &-#{$i} div:nth-child(n + #{$i + 1})::before {
      ??: none;
    }
  }
}
```